internal/services/mssql/mssql_database_resource.go

```diff
@@ -488,133 +492,175 @@ func resourceMsSqlDatabaseRead(d *pluginsdk.ResourceData, meta interface{}) erro
 
-		d.Set("ledger_enabled", ledgerEnabled)
-	}
+			if props.Collation != nil {
+				d.Set("collation", props.Collation)
+			}
 
-	securityAlertPolicy, err := securityAlertPoliciesClient.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
-	if err == nil {
-		if err := d.Set("threat_detection_policy", flattenMsSqlServerSecurityAlertPolicy(d, securityAlertPolicy)); err != nil {
-			return fmt.Errorf("setting `threat_detection_policy`: %+v", err)
-		}
-	}
+			if props.ElasticPoolId != nil {
+				elasticPoolId = pointer.From(props.ElasticPoolId)
+			}
 
-	geoBackupPolicy := true
+			if props.MinCapacity != nil {
+				minCapacity = pointer.From(props.MinCapacity)
+			}
 
-	// DW SKU's do not currently support LRP and do not honour normal SRP operations
-	if !strings.HasPrefix(skuName, "DW") {
-		longTermPolicy, err := longTermRetentionClient.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
-		if err != nil {
-			return fmt.Errorf("retrieving Long Term Retention Policies for %s: %+v", id, err)
-		}
-		if err := d.Set("long_term_retention_policy", helper.FlattenLongTermRetentionPolicy(&longTermPolicy, d)); err != nil {
-			return fmt.Errorf("setting `long_term_retention_policy`: %+v", err)
-		}
+			if props.HighAvailabilityReplicaCount != nil {
+				d.Set("read_replica_count", props.HighAvailabilityReplicaCount)
+			}
 
-		shortTermPolicy, err := shortTermRetentionClient.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
-		if err != nil {
-			return fmt.Errorf("retrieving Short Term Retention Policies for %s: %+v", id, err)
-		}
+			if props.RequestedBackupStorageRedundancy != nil {
+				d.Set("storage_account_type", string(pointer.From(props.RequestedBackupStorageRedundancy)))
+			}
 
-		if err := d.Set("short_term_retention_policy", helper.FlattenShortTermRetentionPolicy(&shortTermPolicy, d)); err != nil {
-			return fmt.Errorf("setting `short_term_retention_policy`: %+v", err)
-		}
-	} else {
-		// DW SKUs need the retention policies zeroing for state consistency
-		zero := make([]interface{}, 0)
-		d.Set("long_term_retention_policy", zero)
-		d.Set("short_term_retention_policy", zero)
+			if props.ZoneRedundant != nil {
+				d.Set("zone_redundant", props.ZoneRedundant)
+			}
 
-		geoPoliciesResponse, err := geoBackupPoliciesClient.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
-		if err != nil {
-			if utils.ResponseWasNotFound(resp.Response) {
-				d.SetId("")
-				return nil
+			if props.ReadScale != nil {
+				d.Set("read_scale", pointer.From(props.ReadScale) == databases.DatabaseReadScaleEnabled)
+			}
+
+			if props.LicenseType != nil {
+				d.Set("license_type", string(pointer.From(props.LicenseType)))
+			} else {
+				// value not returned, try to set from state
+				d.Set("license_type", d.Get("license_type").(string))
 			}
-			return fmt.Errorf("retrieving Geo Backup Policies for %s: %+v", id, err)
+
+			if props.MaxSizeBytes != nil {
+				d.Set("max_size_gb", int32((*props.MaxSizeBytes)/int64(1073741824)))
+			}
```

Comment:
We should be able to make this:

```suggestion
			maxSizeGb := 0
			if props.MaxSizeBytes != nil {
				maxSizeGb = *props.MaxSizeBytes/int64(1073741824)
			}
			d.Set("max_size_gb", maxSizeGb)
```

Reply:
No, actually we can't, else the compiler complains `cannot use *props.MaxSizeBytes / int64(1073741824) (value of type int64) as int value in assignmentcompiler`